fcn_base_node.py

```diff
@@ -660,7 +660,7 @@ def getSocketPosition(self, index: int, position: int, num_out_of: int = 1) -> [
         # Calculates default socket position
         x, y = super().getSocketPosition(index, position, num_out_of)
 
-        if hasattr(self.content, "input_labels"):
+        if hasattr(self.content, "input_labels") and index < len(self.content.input_labels):
```

Comment:
@FlachyJoe Hi, I had initially taken this condition from you, but found that it prevents the correct calculation of the socket positions if there is only one socket on the respective node side. Without the extra condition ` and index < len(self.content.input_labels)` everything works  without problems. What is it good for?

Reply:
This is a necessary workaround because of the label initialization failure. I have to check the whole initialization chain.

The devil's error:  fcn_base_node.py, line **666** :smiling_imp: : 
```
20:43:56  Traceback (most recent call last):
20:43:56    File ".local/lib/python3.10/site-packages/nodeeditor/node_graphics_view.py", line 136, in dropEvent
20:43:56      for callback in self._drop_listeners: callback(event)
20:43:56    File ".local/share/FreeCAD/Mod/fc_nodes/./nodes/sub_patch.py", line 137, in onChange
20:43:56      self.resetWidgets()
20:43:56    File ".local/share/FreeCAD/Mod/fc_nodes/./nodes/sub_patch.py", line 156, in resetWidgets
20:43:56      self.initSockets(inputs, outputs)
20:43:56    File ".local/share/FreeCAD/Mod/fc_nodes/./fcn_base_node.py", line 735, in initSockets
20:43:56      socket: FCNSocket = self.__class__.Socket_class(
20:43:56    File ".local/share/FreeCAD/Mod/fc_nodes/./fcn_base_node.py", line 250, in __init__
20:43:56      super().__init__(node, index, position, socket_color, multi_edges, count_on_this_node_side, is_input)
20:43:56    File ".local/lib/python3.10/site-packages/nodeeditor/node_socket.py", line 73, in __init__
20:43:56      self.setSocketPosition()
20:43:56    File ".local/lib/python3.10/site-packages/nodeeditor/node_socket.py", line 106, in setSocketPosition
20:43:56      self.grSocket.setPos(*self.node.getSocketPosition(self.index, self.position, self.count_on_this_node_side))
20:43:56    File ".local/share/FreeCAD/Mod/fc_nodes/./fcn_base_node.py", line 666, in getSocketPosition
20:43:56      elem: QWidget = self.content.input_labels[index]
20:43:56  IndexError: list index out of range
```